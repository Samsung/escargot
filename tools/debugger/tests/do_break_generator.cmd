b do_break_generator.js:21
c
bt
c
bt
c
bt
c
