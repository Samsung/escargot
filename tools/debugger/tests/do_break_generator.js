/*
 * Copyright (c) 2021-present Samsung Electronics Co., Ltd
 *
 *  This library is free software; you can redistribute it and/or
 *  modify it under the terms of the GNU Lesser General Public
 *  License as published by the Free Software Foundation; either
 *  version 2.1 of the License, or (at your option) any later version.
 *
 *  This library is distributed in the hope that it will be useful,
 *  but WITHOUT ANY WARRANTY; without even the implied warranty of
 *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 *  Lesser General Public License for more details.
 *
 *  You should have received a copy of the GNU Lesser General Public
 *  License along with this library; if not, write to the Free Software
 *  Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301
 *  USA
 */

function f1() {
    return 7
}

function f2() {
    return f1()
}

function f3() {
    return f2()
}

function* g() {
    yield f1()
    yield f2()
    yield f3()
}

var it = g()
it.next()

function h() {
  it.next()
}
h()

it.next()
