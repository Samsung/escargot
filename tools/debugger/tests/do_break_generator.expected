Connecting to: localhost:6501
Connection created!!!
Stopped at tools/debugger/tests/do_break_generator.js:38
(escargot-debugger) b do_break_generator.js:21
Breakpoint 1 at tools/debugger/tests/do_break_generator.js:21 (in f1() at line:20, col:1)
(escargot-debugger) c
Stopped at breakpoint:1 tools/debugger/tests/do_break_generator.js:21 (in f1() at line:20, col:1)
(escargot-debugger) bt
tools/debugger/tests/do_break_generator.js:21:5 [depth:0] (in f1)
tools/debugger/tests/do_break_generator.js:39:1 [depth:3]
(escargot-debugger) c
Stopped at breakpoint:1 tools/debugger/tests/do_break_generator.js:21 (in f1() at line:20, col:1)
(escargot-debugger) bt
tools/debugger/tests/do_break_generator.js:21:5 [depth:0] (in f1)
tools/debugger/tests/do_break_generator.js:25:12 [depth:1] (in f2)
tools/debugger/tests/do_break_generator.js:42:3 [depth:4] (in h)
tools/debugger/tests/do_break_generator.js:44:1 [depth:5]
(escargot-debugger) c
Stopped at breakpoint:1 tools/debugger/tests/do_break_generator.js:21 (in f1() at line:20, col:1)
(escargot-debugger) bt
tools/debugger/tests/do_break_generator.js:21:5 [depth:0] (in f1)
tools/debugger/tests/do_break_generator.js:25:12 [depth:1] (in f2)
tools/debugger/tests/do_break_generator.js:29:12 [depth:2] (in f3)
tools/debugger/tests/do_break_generator.js:46:1 [depth:5]
(escargot-debugger) c
Connection closed.
